builder/vsphere/supervisor/step_validate_publish.go

```diff
@@ -0,0 +1,134 @@
+//go:generate packer-sdc struct-markdown
+//go:generate packer-sdc mapstructure-to-hcl2 -type ValidatePublishConfig
+
+package supervisor
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	imgregv1a1 "github.com/vmware-tanzu/image-registry-operator-api/api/v1alpha1"
+	vmopv1alpha1 "github.com/vmware-tanzu/vm-operator/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	StateKeyPublishLocationName = "publish_location_name"
+
+	vmPubFeatureNotEnabledMsg = "WCP_VM_Image_Registry feature not enabled"
+)
+
+type ValidatePublishConfig struct {
+	// Name of the location (e.g. vSphere content library) where the source VM will be published.
```

Comment:
In the foreseeable future, this can only be a Content Library defined in the [image-registry-operator API](https://github.com/vmware-tanzu/image-registry-operator-api/blob/main/api/v1alpha1/contentlibrary_types.go#L169). It's a K8s resource representation of the vSphere Content Library. Let me update the doc to clarify this.

Reply:
Done.